### PR TITLE
[RW-6443][risk=no] Adding validation for cope survey versions

### PR DESCRIPTION
**Pull request by freemabd**
Adding validation for cope survey versions

### CODE REVIEW

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh [hunk 1]
```diff
@@ -61,14 +70,21 @@ if gsutil -m cp gs://$BUCKET/$BQ_DATASET/$CSV_HOME_DIR/*.csv $TEMP_FILE_DIR
...
+    $PREP_CONCEPT_RELATIONSHIP)
       tableName=${file%.*}
       if [[ $firstColumn == id || \
             $firstColumn == ancestor_id || \
             $firstColumn == parent || \
             $firstColumn == concept_id || \
             $firstColumn == concept_id_1 || \
-            $firstColumn == bq_dataset ]];
+            $firstColumn == bq_dataset || \
+            $firstColumn == survey_version_concept_id ]];
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/validate-prerequisites-exist.sh:87]: Added carriage returns for readability

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh [hunk 2]
```diff
@@ -107,19 +125,32 @@ fi
...
-if [[ $cdrDate != 1 ]];
-then
-  echo "CDR cutoff date doesn't exist in $BQ_PROJECT.$BQ_DATASET.prep_cdr_date!"
-  exit 1
-fi
-
-# Purge all backup csv files except for the last 10 versions
 if [[ $CDR_VERSION != $PREP_TABLE_RUN ]];
 then
+
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/validate-prerequisites-exist.sh:130]: Make sure these validate checks don't happen on just building the prep tables run.

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh [hunk 3]
```diff
@@ -107,19 +125,32 @@ fi
...
+  echo "Validating that a CDR cutoff date exists..."
+  q="select count(*) as count from \`$BQ_PROJECT.$BQ_DATASET.prep_cdr_date\` where bq_dataset = '$BQ_DATASET'"
+  cdrDate=$(bq --quiet --project_id=$BQ_PROJECT query --nouse_legacy_sql "$q" | tr -dc '0-9')
+  if [[ $cdrDate != 1 ]];
+  then
+    echo "CDR cutoff date doesn't exist in $BQ_PROJECT.$BQ_DATASET.prep_cdr_date!"
+    exit 1
+  fi
+
+  # Validate that all survey version exist
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/validate-prerequisites-exist.sh:141]: This is the only real code change since the rest is just carriage returns/formatting.

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh [hunk 4]
```diff
@@ -117,9 +135,22 @@ then
   exit 1
 fi
 
-# Purge all backup csv files except for the last 10 versions
 if [[ $CDR_VERSION != $PREP_TABLE_RUN ]];
 then
+
+  # Validate that all survey version exist
+  echo "Validating that all survey versions exist..."
+  q="select count(*) as count from \`$BQ_PROJECT.$BQ_DATASET.cb_survey_version\`
```
Comment (NehaBroad) [inline api/db-cdr/generate-cdr/validate-prerequisites-exist.sh:143]: nit: can we change the name of the query variable?

**Review by NehaBroad — APPROVED** [review]

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh [hunk 5]
```diff
@@ -61,14 +70,21 @@ if gsutil -m cp gs://$BUCKET/$BQ_DATASET/$CSV_HOME_DIR/*.csv $TEMP_FILE_DIR
           gsutil cp $TEMP_FILE_DIR/$file.gz gs://$BUCKET/$BQ_DATASET/$CSV_HOME_DIR/backup/"$timestamp"_"$file".gz
         fi
       ;;
-    $PREP_CDR_DATE|$PREP_CRITERIA|$PREP_CRITERIA_ANCESTOR|$PREP_CLINICAL_TERMS|$PREP_CONCEPT|$PREP_CONCEPT_RELATIONSHIP)
+    $CB_SURVEY_VERSION | \
```
Comment (NehaBroad) [inline api/db-cdr/generate-cdr/validate-prerequisites-exist.sh:73]: Not part of this PR but just a note : Maybe we can create a variable like ALL_FILES to store this subset of file names as the list is too long now

File: api/db-cdr/generate-cdr/validate-prerequisites-exist.sh (hunk 4 above)
Comment (freemabd) [reply api/db-cdr/generate-cdr/validate-prerequisites-exist.sh]: Done.
